[TiwKangXu] ip

TiwKangXu opened the pull request:
# DukePro
> "Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :slightly_smiling_face:

and it is **FREE**! (~~No, it would not be forever free~~)

Features:

- [x] Managing tasks.
- [ ] Managing deadlines. *(coming soon)*
- [ ] Reminders. *(coming soon)*

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method: 

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

TiwKangXu committed:
Update Duke.java

TiwKangXu committed:
Complete Level-0

TiwKangXu committed:
Complete Level-1

TiwKangXu committed:
Complete Level-2

TiwKangXu committed:
Complete Level-3

TiwKangXu committed:
Complete Level-4

TiwKangXu committed:
Complete A-TextUiTesting

TiwKangXu committed:
Complete Level-5

TiwKangXu committed:
Complete Level-6

TiwKangXu committed:
Complete A-Enums

TiwKangXu committed:
Add Documentation

TiwKangXu committed:
Add documentation

TiwKangXu committed:
Draft Level-7

TiwKangXu committed:
Complete Level-7

TiwKangXu committed:
Complete Level-8

TiwKangXu committed:
Complete A-MoreOOP

TiwKangXu committed:
Complete Level-8

TiwKangXu committed:
Complete A-Packages

TiwKangXu committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

TiwKangXu committed:
Complete A-JUnit

TiwKangXu committed:
Update A-JUnit

TiwKangXu committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-JavaDoc

TiwKangXu committed:
Merge branch 'branch-A-JavaDoc'

TiwKangXu committed:
Complete A-CodingStandard

TiwKangXu committed:
Complete Level-9

TiwKangXu committed:
Merge branch 'branch-Level-9'

TiwKangXu committed:
Update Parser

TiwKangXu committed:
Switch to Java 11 from Java 17

TiwKangXu committed:
Add documentation.

TiwKangXu committed:
Update documentation

TiwKangXu committed:
Complete A-CheckStyle

TiwKangXu committed:
Remove all JavaFX Dependencies. Clean repository.

TiwKangXu committed:
Complete JavaFX setup

TiwKangXu committed:
Complete adding fxml files.

TiwKangXu committed:
Complete Level-10

TiwKangXu committed:
Add invalid argument exception for mark and unmark

siqirua commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,48 @@
+        storage.load();
+    }
+
+    /**
+     * Bridges GUI and Program. Takes in user command scanned by MainWindow and parses it.
+     * @param reply
+     * @return chatbot's reply
+     */
```
```suggestion
    /**
     * Bridges GUI and Program. Takes in user command scanned by MainWindow and parses it.
     * 
     * @param reply
     * @return chatbot's reply
     */
```
Would it be better to put an empty line between description and parameter section?

siqirua commented on src/main/java/duke/exception/DukeInvalidDateFormatException.java:
```diff
@@ -0,0 +1,10 @@
+ * Handles invalid date.
+ */
+public class DukeInvalidDateFormatException extends DukeException {
+    public String toString() {
```
```suggestion
    @Override
    public String toString() {
```
Would it be better to put an Override?

TiwKangXu replied:
Thanks for the suggestion.

siqirua commented on src/main/java/duke/exception/DukeInvalidSavedToFileLineType.java:
```diff
@@ -0,0 +1,12 @@
+/**
+ * Handles unknown command entered.
+ */
+public class DukeInvalidSavedToFileLineType extends DukeException {
```
```suggestion
public class DukeInvalidSavedToFileLineTypeException extends DukeException {
```
Do you think it would be better if you can state "Exception" in the variable name to show the meaning more explicitly?

TiwKangXu replied:
Yes! Thanks for the suggestion.

siqirua commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,50 @@
+                statusAndTaskContent, from.format(formatter), to.format(formatter));
+    }
+
+    /**
+     * Creates Event task.
+     * @param status
+     * @param description
+     * @param from
+     * @param to
+     * @return
+     */
```
Maybe can elaborate more on the parameters?

siqirua commented on src/main/java/duke/task/ToDo.java:
```diff
@@ -0,0 +1,34 @@
+     * @param description
+     * @return
+     */
+    public static ToDo create(String status, String description) {
+        ToDo task = new ToDo(description);
+        if (status == "1") {
+            task.mark();
+        }
+        return task;
+    }
```
Maybe can consider implement another constructor for this?

siqirua reviewed (commented):
Overall good job done! The code standard is well followed except that the JavaDoc can be further improved. I like how you write your method name. Keep it up!

shuyangk commented on src/main/java/duke/exception/DukeException.java:
I like how you created different classes for the different exceptions!

shuyangk commented on src/main/java/duke/Duke.java:
I like how neat you made your Duke file!

shuyangk commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,220 @@
+    /**
+     * Exits chatbot.
+     */
+    private String exit() {
+        String guiTalk = "  Bye~ Hope to see you again soon! >w<";
+        return guiTalk;
```
Perhaps the strings to be displayed to the user could be placed within the Ui class?

TiwKangXu replied:
Thanks for this suggestion! I've edited it.

shuyangk commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,50 @@
+/**
+ * Event task.
+ */
+public class Event extends Task {
+    private LocalDate from;
+    private LocalDate to;
```
I like how you named your variables!

shuyangk commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,60 @@
+ */
+public class Task {
+    protected DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy");
+    private boolean status;
```
Perhaps you could consider changing this boolean name to something more intuitive (e.g. isDone)?

TiwKangXu replied:
Thanks for this suggestion!

shuyangk reviewed (commented):
Overall I found your code easy to read, and there were no coding standard violations. The encapsulation is very clear, however some variable names might be slightly unclear.

TiwKangXu committed:
Complete A-CodeQuality

Minor flaws in achieving good code quality.

Good code quality enhances readability.

Refine javadocs, implement more OOP.

More OOP breaks task smaller.

TiwKangXu committed:
Merge pull request #1 from TiwKangXu/branch-A-CodeQuality

Complete A-CodeQuality

TiwKangXu committed:
Complete A-Assertions

Current code lack assertions.

Assertions define assumptions about the program state so that the runtime can verify them.

Added assertions in Parser.java.

TiwKangXu committed:
Merge branch 'master' into branch-A-Assertions

TiwKangXu committed:
Merge pull request #2 from TiwKangXu/branch-A-Assertions

Complete A-Assertions

TiwKangXu committed:
Start A-CI

Current project is not set up with Continuous Integration.

Continuous Integration and Continuous Delivery enhance efficiency for shipping software.

GitHub Actions will run the workflow (as per the .yml file) every time certain project events are triggered (e.g., when a PR is updated, or the master branch is updated).

TiwKangXu committed:
Complete C-DetectDuplicates

TiwKangXu committed:
Complete A-Streams

TiwKangXu committed:
Improve Code Quality

Current project has minor check-style violations.

Enhanced code quality improve readability.

Changes on header comments for functions, library imports, and removing
dead codes.

TiwKangXu committed:
Fix bug for load and save

TiwKangXu committed:
Update README.md

TiwKangXu committed:
Create Ui.png

TiwKangXu committed:
Update User Guide

TiwKangXu committed:
Update User Guide in docs/Readme.md

TiwKangXu committed:
Update Readme.md

TiwKangXu committed:
Update README.md

TiwKangXu committed:
Switch Java 17 to Java 11

Create .java-version

TiwKangXu committed:
Fix build.gradle

TiwKangXu committed:
Update build.gradle

TiwKangXu committed:
Update build.gradle

TiwKangXu committed:
Update build.gradle

TiwKangXu committed:
Update build.gradle
